backend/app.hopps.az-document-ai/src/main/java/app/hopps/AzureAiService.java

```diff
@@ -1,73 +1,98 @@
 package app.hopps;
 
-import app.hopps.model.InvoiceData;
-import app.hopps.model.ReceiptData;
-import com.azure.ai.documentintelligence.DocumentIntelligenceClient;
-import com.azure.ai.documentintelligence.DocumentIntelligenceClientBuilder;
-import com.azure.ai.documentintelligence.models.AnalyzeDocumentRequest;
+import app.hopps.auth.FinRestClientImpl;
+import app.hopps.commons.DocumentData;
+import app.hopps.commons.InvoiceData;
+import app.hopps.commons.ReceiptData;
+import app.hopps.model.InvoiceDataHelper;
+import app.hopps.model.ReceiptDataHelper;
 import com.azure.ai.documentintelligence.models.AnalyzeResult;
-import com.azure.ai.documentintelligence.models.AnalyzeResultOperation;
 import com.azure.ai.documentintelligence.models.Document;
-import com.azure.core.credential.AzureKeyCredential;
-import com.azure.core.util.polling.SyncPoller;
+import io.quarkus.oidc.client.OidcClient;
+import io.quarkus.oidc.client.runtime.TokensHelper;
 import jakarta.enterprise.context.ApplicationScoped;
 import jakarta.inject.Inject;
 import org.eclipse.microprofile.config.inject.ConfigProperty;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.net.URISyntaxException;
 import java.net.URL;
+import java.time.Duration;
 import java.util.List;
 
 @ApplicationScoped
 public class AzureAiService {
 
-    private final Logger LOGGER = LoggerFactory.getLogger(AzureAiService.class);
+    private static final Logger LOG = LoggerFactory.getLogger(AzureAiService.class);
+
+    private final AzureDocumentConnector azureDocumentConnector;
 
     @ConfigProperty(name = "app.hopps.az-document-ai.azure.invoiceModelId")
     String invoiceModelId;
 
     @ConfigProperty(name = "app.hopps.az-document-ai.azure.receiptModelId")
     String receiptModelId;
 
+    private final OidcClient oidcClient;
+    private final TokensHelper tokensHelper = new TokensHelper();
```

Comment:
Einfach nur `Tokens` injecten. Die sind immer aktuell.

Reply:
Wenn ich das versuche bekomme ich immer ein Fehler:
`RequestScoped context was not active when trying to obtain a bean instance for a client proxy of PRODUCER_METHOD bean [class=io.quarkus.oidc.client.runtime.TokensProducer, id=nu9k6HilQ8JNgG6ZFZTSp_rDHzY]`

Reply:
Klar. Das findet ja auch außerhalb von einem Request statt, weil Kafka :) 

Dann lassen wir es so.